### ISSUE TITLE
Would a `Val::PhysicalPx` be useful?

### ISSUE THREAD

**Issue by ickshonpe** [bevyengine/bevy#8135]
## What problem does this solve or what need does it fill?

`Val::Px` represents a length in logical pixels.
`Val::PhysicalPx` (or whatever name it was given) would represent a length in physical pixels.
It would be easy to implement, but I'm not sure how useful it would be.

**Comment by mockersf** [comment]
I don't think so, it would promote a bad pattern of not caring about the user scale factor.

Also, UI already has an inner scaling mechanism

**Comment by mirsella** [comment]
We ran into a practical use for this with decorative hairline borders. At fractional scale factors, `Val::Px(1.)` becomes a fractional physical width, so opposite edges can rasterize differently. A physical-pixel unit would keep those details one target pixel without disabling UI scaling for the rest of the layout. It should probably be used sparingly, but it seems useful for borders and other raster-sensitive details.
